src/lib/optimizer/strategy/predicate_pushdown_rule.cpp

```diff
@@ -13,26 +13,41 @@ namespace opossum {
 
 std::string PredicatePushdownRule::name() const { return "Predicate Pushdown Rule"; }
 
+void push_down(const std::shared_ptr<AbstractLQPNode>& node, std::shared_ptr<AbstractLQPNode> input) {
```

Comment:
Can you pick parameter names that better reflect which node is moved and which is the new parent?

Reply:
Also, pls make this a member function or put it into an anonymous namespace